File: quick-build/corpus/apps/hello-java/src/main/java/org/appdevforall/cotg/corpus/hellojava/MainActivity.java
package org.appdevforall.cotg.corpus.hellojava;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Gravity;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

public class MainActivity extends Activity {

	public static final int ID_GREETING = 2001;

	private final Greeter greeter = new Greeter();

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		LinearLayout root = new LinearLayout(this);
		root.setOrientation(LinearLayout.VERTICAL);
		root.setGravity(Gravity.CENTER);

		TextView greetingView = new TextView(this);
		greetingView.setId(ID_GREETING);
		greetingView.setText(greeter.greet(getString(R.string.greeting_name)));
		root.addView(greetingView);

		Button detailButton = new Button(this);
		detailButton.setText(getString(R.string.detail_label));
		detailButton.setOnClickListener(v -> startActivity(new Intent(this, DetailActivity.class)));
		root.addView(detailButton);

		setContentView(root);
	}
}

File: quick-build/corpus/apps/hello-kotlin/src/main/java/org/appdevforall/cotg/corpus/hellokotlin/MainActivity.kt
package org.appdevforall.cotg.corpus.hellokotlin

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.Gravity
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView

class MainActivity : Activity() {
	private val greeter = Greeter()

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)

		val root = LinearLayout(this)
		root.orientation = LinearLayout.VERTICAL
		root.gravity = Gravity.CENTER

		val greetingView = TextView(this)
		greetingView.id = ID_GREETING
		greetingView.text = greeter.greet(getString(R.string.greeting_name))
		root.addView(greetingView)

		val detailButton = Button(this)
		detailButton.text = getString(R.string.detail_label)
		detailButton.setOnClickListener {
			startActivity(Intent(this, DetailActivity::class.java))
		}
		root.addView(detailButton)

		setContentView(root)
	}

	companion object {
		const val ID_GREETING = 1001
	}
}
